src/main/java/com/hartwig/snpcheck/SnpCheck.java

```diff
@@ -61,45 +65,71 @@ public SnpCheck(final RunApi runs, final SampleApi samples, final Bucket snpchec
         this.validatedTopicPublisher = validatedTopicPublisher;
         this.objectMapper = objectMapper;
         this.labPendingBuffer = new LabPendingBuffer(this, Executors.newScheduledThreadPool(1), TimeUnit.HOURS, 1);
+        this.pendingEvents = new ArrayList<>();
+    }
+
+    private Optional<List<PipelineStaged>> enqueueEvent(final PipelineStaged event) {
```

Comment:
I think functionally we can also just run the snpcheck 3 times and publish the 3 events, its basically the same problem as the research vs diagnostic. 

In the end, probably we need to revisit the model of having the run split in 3.It's causing a lot of complexity and in the end we're taking an atomic things and splitting it, with the only real reason to persist it to different places. That could be changed to just group the files within a single event or something.